navigation/MainNavigator.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {
+  FunctionComponent,
+  ReactElement,
+  useCallback,
+  useState,
+} from "react";
+import { LayoutAnimation, Platform, StyleSheet, UIManager } from "react-native";
+import { View, Text } from "../components/Themed";
+import EpisodePredictionWrapper from "../screens/EpisodePrediction";
+import MainScreen from "../screens/MainScreen";
+import CameraFlowNavigator from "./CameraFlowNavigator";
+import {
+  MainNavigationContext,
+  NavigationScreens,
+  NavigationState,
+} from "./MainNavigationContext";
+import EpisodeNavigator from "./MainStack";
+
+const MainNavigator: FunctionComponent = () => {
+  const [navigationState, setNavigationState] = useState<NavigationState>({
+    type: NavigationScreens.intro,
+  });
+
+  if (
+    Platform.OS === "android" &&
+    UIManager.setLayoutAnimationEnabledExperimental
+  ) {
+    UIManager.setLayoutAnimationEnabledExperimental(true);
+  }
+
+  // Get the correct screen from the given state
+  const getScreen = (
+    navigationState: NavigationState
+  ): ReactElement | undefined => {
+    if (navigationState.type == NavigationScreens.intro) {
+      return <MainScreen />;
+    } else if (navigationState.type === "recordEpisodes") {
+      // Putting a default value into the overlay creator until we have the actual value
+      return (
+        <CameraFlowNavigator
+          objects={navigationState.episodes}
+          overlayCreator={(episode) => (
+            <Text style={{ color: "white" }}>{episode.name}</Text>
+          )}
+        />
+      );
+    } else if (navigationState.type == NavigationScreens.createEpisode) {
+      return <EpisodeNavigator />;
+    } else if (navigationState.type == NavigationScreens.predictions) {
+      return <EpisodePredictionWrapper />;
+    } else {
+      return undefined;
+    }
```

Comment:
So this is for the flow of a certain day, once we start thinking about day 2 and 3 and the "flow" is a little different, does this function become a series of a bunch of if-else's? Like the episode creation recording won't flow into episode prediction on day 3 at all.

Reply:
Yep exactly!